### PR TITLE
Functional problems fixes. Part 1

### PR DESCRIPTION
**Pull request by mandrono**

### CODE REVIEW

**Comment by mandrono** [comment]
@nshchego take a look please

File: inference-engine/src/mkldnn_plugin/mkldnn_node.cpp [hunk 1]
```diff
@@ -135,6 +135,7 @@ static const InferenceEngine::details::caseless_unordered_map<std::string, Type>
 //        { "Softmax", SoftMax },
 //        { "SoftMax", SoftMax },
         { "Split", Split },
+        { "VariadicSplit", Split },
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/mkldnn_node.cpp:138]: Duplicates line140

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp [hunk 2]
```diff
@@ -217,13 +217,11 @@ void MKLDNNGraph::Replicate(const CNNNetwork &network, const MKLDNNExtensionMana
     std::map<std::shared_ptr<ngraph::Node>, MKLDNNNodePtr> op2node;
     std::deque<ngraph::Output<ngraph::Node>> unusedOutputs;  // nodes which has no consumers (output or just unused)
 
-    auto getParentPort = [](const std::shared_ptr<ngraph::Node> op, const std::shared_ptr<ngraph::Node> parentOp) -> int {
+    auto getParentPort = [](const std::shared_ptr<ngraph::Node> op, const std::shared_ptr<ngraph::Node> parentOp, const size_t port) -> int {
         // TODO [NM]: do we have a better way how to determine parent port?
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp:221]: Do we still need this comment?

File: inference-engine/src/mkldnn_plugin/mkldnn_memory.cpp [hunk 3]
```diff
@@ -701,6 +693,19 @@ MKLDNNMemoryDesc::operator InferenceEngine::TensorDesc() const {
...
+    }
+
+    // order of outer dims. In case of IOhw_ will be {1, 0, 2, 3}
+    std::vector<size_t> outer_order(outer_ndims);
+    std::iota(outer_order.begin(), outer_order.end(), 0);
+    std::sort(outer_order.begin(), outer_order.end(),
+              [&blk_desc, &outer_block_dims] (size_t ind_l, size_t ind_r) {
+        return (blk_desc.strides[ind_l] > blk_desc.strides[ind_r]) ||
+               (blk_desc.strides[ind_l] == blk_desc.strides[ind_r] && outer_block_dims[ind_l] > outer_block_dims[ind_r]);
+    });
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/mkldnn_memory.cpp:708]: Could you add test case for that?

File: inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp [hunk 4]
```diff
@@ -339,6 +342,8 @@ static void Transformation(CNNNetwork& clonedNetwork, const Config& conf) {
     postLPTPassManager.run_passes(nGraphFunc);
 
     ConvertToCPUSpecificOpset(nGraphFunc);
+
+    // ngraph::pass::VisualizeTree("/home/maximandronov/test_repo/openvino/models/DEBUG_INFO/after.png").run_on_function(nGraphFunc);
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp:346]: Remove?

File: inference-engine/src/mkldnn_plugin/ngraph_transformations/convert_matmul_to_fc_or_gemm.cpp [hunk 5]
```diff
@@ -170,7 +175,33 @@ MKLDNNPlugin::ConvertMatMulToGemm::ConvertMatMulToGemm() {
...
+            // For example {S} will be reshaped to {S, 1}.
+            fc_input_b = std::make_shared<ngraph::opset1::Unsqueeze>(fc_input_b,
+                ngraph::opset1::Constant::create(ngraph::element::i64, ngraph::Shape{1}, {1}));
+            shape_b = fc_input_b.get_shape();
+            new_ops.push_back(fc_input_b.get_node_shared_ptr());
+            // For 1D inputs transpose flag is expected to always act like `false`
+            matmul->set_transpose_b(false);
+        }
+
         // WA for IE that Gemm must have inputs with the same length.
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/ngraph_transformations/convert_matmul_to_fc_or_gemm.cpp:201]: Seems, Gemm node was removed. Why do we need this WA?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_fullyconnected_node.cpp [hunk 6]
```diff
@@ -273,7 +272,7 @@ void MKLDNNFullyConnectedNode::createDescriptor(const std::vector<InferenceEngin
 MKLDNNMemoryDesc MKLDNNFullyConnectedNode::getSrcMemDesc(mkldnn::primitive_desc_iterator &primitive_desc_it, size_t idx) {
     InferenceEngine::TensorDesc desc = idx > 0 ? MKLDNNMemoryDesc(primitive_desc_it.weights_desc(idx - 1))
                                                : MKLDNNMemoryDesc(primitive_desc_it.src_desc(idx));
-
+    // std::cout << "IDX: " << idx << " " << getName() << std::endl;
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_fullyconnected_node.cpp:275]: Remove

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 7]
```diff
@@ -59,10 +59,10 @@ static TensorDesc makeChannelBlockedTensorDesc(const Precision& precision, const
 
 bool MKLDNNSplitNode::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
     try {
-        if (!MKLDNNPlugin::one_of(op->get_type_info(),
-                ngraph::op::v1::Split::type_info,
-                ngraph::op::v1::VariadicSplit::type_info)) {
-            errorMessage = "Node is not an instance of the Split operation.";
+        auto splitOp = ngraph::as_type_ptr<const ngraph::op::v1::Split>(op);
+        auto variadSplitOp = ngraph::as_type_ptr<const ngraph::op::v1::VariadicSplit>(op);
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:63]: Why don't you believe to type_info?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp [hunk 8]
```diff
@@ -147,26 +156,21 @@ void MKLDNNSplitNode::initSupportedPrimitiveDescriptors() {
...
         config.inConfs[1].desc.setDims({1});
         config.inConfs[1].desc.setPrecision(axisPrecision);
-        if (!splitLengthsDims.empty()) {
-            config.inConfs.resize(3);
-            config.inConfs[2].inPlace = -1;
-            config.inConfs[2].constant = true;
-            config.inConfs[2].desc.setDims(splitLengthsDims);
-            config.inConfs[2].desc.setPrecision(splitLengthsPrecision);
+        if (inputs_num == 3) {
+            config.inConfs[2].desc = TensorDesc(axisPrecision, SizeVector{outDims.size()}, TensorDesc::getLayoutByDims(SizeVector{outDims.size()}));
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:173]: config.inConfs[2].constant = true;

File: inference-engine/src/mkldnn_plugin/mkldnn_node.cpp (hunk 1 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/mkldnn_node.cpp:138]: removed

File: inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp (hunk 2 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/mkldnn_graph.cpp]: I think not

File: inference-engine/src/mkldnn_plugin/mkldnn_memory.cpp (hunk 3 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/mkldnn_memory.cpp:708]: the problem was initially reproduced for convolution so was added case with specific weights layout also tests ```QuantConvBackpropDataLayerTest``` have the same problem

File: inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp (hunk 4 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/mkldnn_plugin.cpp]: removed

File: inference-engine/src/mkldnn_plugin/ngraph_transformations/convert_matmul_to_fc_or_gemm.cpp (hunk 5 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/ngraph_transformations/convert_matmul_to_fc_or_gemm.cpp:201]: we just renamed Gemm -> MatMul but we still need this

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_fullyconnected_node.cpp (hunk 6 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_fullyconnected_node.cpp]: removed

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp (hunk 7 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp]: switched to type info

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp (hunk 8 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_split_node.cpp:171]: added

**Comment by mandrono** [comment]
@nshchego could you take a look one more time?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.h [hunk 9]
```diff
@@ -44,6 +44,16 @@ class MKLDNNConvolutionNode : public MKLDNNNode {
...
 
+    void setStride(const std::vector<ptrdiff_t> &_stride) { stride = _stride; }
+    void setDilation(const std::vector<ptrdiff_t> &_dilation) { dilation = _dilation; }
+    void setPaddingL(const std::vector<ptrdiff_t> &_paddingL) { paddingL = _paddingL; }
+    void setPaddingR(const std::vector<ptrdiff_t> &_paddingR) { paddingR = _paddingR; }
+
+    std::vector<ptrdiff_t> getStride() { return stride; }
+    std::vector<ptrdiff_t> getDilation() { return dilation; }
+    std::vector<ptrdiff_t> getPaddingL() { return paddingL; }
+    std::vector<ptrdiff_t> getPaddingR() { return paddingR; }
```
Comment (nshchego) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.h:55]: Perhaps 'const vector<..>&' ?

**Review by nshchego — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.h (hunk 9 above)
Comment (mandrono) [reply inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.h]: fixed

**Pull request merged**